include/modules/dwm.hpp

```diff
@@ -0,0 +1,371 @@
+#pragma once
+
+#include <dwmipcpp/connection.hpp>
+
+#include "modules/meta/event_module.hpp"
+#include "modules/meta/input_handler.hpp"
+
+POLYBAR_NS
+
+namespace modules {
+  class dwm_module : public event_module<dwm_module>, public input_handler {
+   public:
+    explicit dwm_module(const bar_settings&, string);
+
+    using tag_mask_t = unsigned int;
+    using window_t = unsigned int;
+
+    /**
+     * Represents the relevant states a tag can have
+     */
+    enum class state_t : uint8_t {
+      FOCUSED,    ///< Monitor is selected and tag is selected, overrides all below
+      URGENT,     ///< Tag is urgent, overrides all below
```

Comment:
When I was originally making the module, I thought that if a tag was focused, it wouldn't be urgent. Now that I think about it, it would be possible to have a tag in focus, but one of its unfocused windows have the urgent hint set. I will fix this.